alcs-api/src/application/application-decision/application-decision.controller.ts

```diff
@@ -43,28 +45,39 @@ export class ApplicationDecisionController {
 
   @Get('/application/:fileNumber')
   @UserRoles(...ANY_AUTH_ROLE)
-  async getAllForApplication(@Param('fileNumber') fileNumber): Promise<{
-    decisions: ApplicationDecisionDto[];
-    codes: ApplicationDecisionOutcomeTypeDto[];
-  }> {
+  async getByFileNumber(
+    @Param('fileNumber') fileNumber,
+  ): Promise<ApplicationDecisionDto[]> {
     const decisions = await this.appDecisionService.getByAppFileNumber(
       fileNumber,
     );
-    const codes = await this.appDecisionService.getCodeMapping();
-    const mappedDecisions = this.mapper.mapArray(
+    return this.mapper.mapArray(
       decisions,
       ApplicationDecision,
       ApplicationDecisionDto,
     );
-    const mappedCodes = this.mapper.mapArray(
-      codes,
-      ApplicationDecisionOutcomeType,
-      ApplicationDecisionOutcomeTypeDto,
-    );
+  }
 
+  @Get('/codes')
+  @UserRoles(...ANY_AUTH_ROLE)
+  async getCodes() {
```

Comment:
With this code here we have a new pattern now. There is CodeService that aggregates all code types and here we have entity-specific types. So do you want to keep these separate? What if we need one of this type on board, should we duplicate it there?

Reply:
As we move away from needing to map the code using the code service to get the label, we might even end up deleting the main CodeService. Now you will only need the full list of codes in places where you can edit them or need to display them all (boards). I made this independent since they are only needed in the create/edit dialog of decisions, you can display them on the board since it will eventually return the full dto without needing the full list.